proxy/src/serverless/backend.rs

```diff
@@ -182,6 +239,10 @@ pub(crate) enum HttpConnError {
     ConnectionClosedAbruptly(#[from] tokio::sync::watch::error::SendError<uuid::Uuid>),
     #[error("could not connection to compute")]
     ConnectionError(#[from] tokio_postgres::Error),
+    #[error("could not connection to compute")]
+    IoConnectionError(#[from] std::io::Error),
```

Comment:
Using `#[from]` with generic errors can easily lead to unrelated errors being wrapped in this variant. That's made worse by not including the inner error in the message.